package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -0,0 +1,358 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Reading trajectory with Chemfiles --- :mod:`MDAnalysis.coordinates.chemfiles`
+=============================================================================
+
+Classes to read and write files using the `chemfiles`_ library. This library
+provides C++ implementation of multiple formats readers and writers, the full
+list if available `here <formats>`_.
+
+.. _chemfiles: https://chemfiles.org
+.. _formats: http://chemfiles.org/chemfiles/latest/formats.html
+
+Classes
+-------
+
+.. autoclass:: ChemfilesReader
+
+.. autoclass:: ChemfilesWriter
+
+"""
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+
+import numpy as np
+from six import string_types
+from distutils.version import LooseVersion
+
+from . import base, core
+
+import chemfiles
+from chemfiles import Trajectory, Frame, Atom, UnitCell, Residue, Topology
+
+
+MIN_CHEMFILES_VERSION = LooseVersion("0.9")
+MAX_CHEMFILES_VERSION = LooseVersion("0.10")
+
+
+def check_chemfiles_version():
```

Comment:
@richardjgowers you want a nicer error message than this? I notice the imports should be in the classes themselves to avoid an error when importing mda without chemfiles.